File: submissions/java-archive-binks/challenge_9.java
static void f(String s) {
        int livesLeft = 2, invul = 0, coins=0;
        boolean big = false;
        for(String ev: s.split(" ")){
            if("".equals(ev)){continue;}
            if(("piranha".equals(ev) || "goomba".equals(ev) || "koopa".equals(ev)) && invul == 0){
                if(big) {big= false;} else {livesLeft--;}
            }
            if("1Up".equals(ev)){
                livesLeft++;
            }
            if("Mushroom".equals(ev)){
                big = true;
            }
            if("Star".equals(ev)){
                invul = 3;
            }
            if("Princess".equals(ev) || "Bowser".equals(ev)){
                System.out.print("WIN");
                return;
            }
            try {
                coins += Integer.parseInt(ev);
            }catch (Exception e){}
            livesLeft += coins/100;
            coins%=100;
            invul = Math.max(0,invul-1);
            if (livesLeft<0){
                System.out.print("GAME OVER");
                return;
            }
        }
        System.out.print("???");
    }

File: submissions/bat/challenge9.rb
def f(s)
$x=s.split(" ")
$m=Hash["c",0,"l",3,"b",0,"i",0]
$r='???'
def e()
if $m["i"]==0;if $m["b"] > 0;$m["b"]=0;else $m["l"]-=1;end
end
end
def c(a)
$m["c"]+=a.to_i
$h=($m["c"]/100).floor
if $h>0;$m["l"]+=$h;$m["c"]-=$h*100
end
end
for i in 0..$x.length
case $x[i]
when '1Up';$m["l"]+=1
when 'Star';$m["i"]=3
when 'Mushroom';$m["b"]=1
when "goomba","koopa","piranha";e()
when "Bowser","Princess";$r="WIN"
else c($x[i])
end
if $m["l"]<1 && $r!='WIN'
$r='GAME OVER';break;end
if $m["i"]>0;$m["i"]-=1;end;end
puts $r
end